Inputs: Improve typing of onChange handlers

alexis-ld opened the pull request:
Refs: #231

**Component**:
Checkbox
Input
SearchInput
TextArea
Toggle

<!-- E.g. 'Layout', 'Table', 'build', 'tests'... -->

**Description**:
Added type for onChange handlers

---

<!-- Declare one or more issues to close once this PR gets merged -->


<!-- If you want to refer to an issue while not closing it, use:

See: #231 

-->

alexis-ld committed:
inputs: Improve typing of onChange handlers

Refs: #231

bert-e commented:
# Hello alexis-ld,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/core-ui/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @carlito-scality

* @nicolas2bert

* @rachedbenmustapha

* @monpote

* @chengyanjin

ChengYanJin commented on stories/input.js:
```diff
@@ -109,13 +109,13 @@ const ExampleInput = () => {
         value={numberValue}
         onChange={(e) => {
-          if (e && e.target) setNumberValue(e.target.value);
+          if (e && e.target) setNumberValue(e.currentTarget.value);
```
I don't think it's needed to check the existence of the event anymore since we added the typing. WDYT?

```suggestion
        setNumberValue(e.currentTarget.value);
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @carlito-scality

* @nicolas2bert

* @rachedbenmustapha

* @monpote

* @chengyanjin





The following reviewers are expecting changes from the author, or must review again:

* @chengyanjin

alexis-ld committed:
Input: Remove unneeded verification

Since typing is enabled we don't need to check that the event has a target property anymore

Refs: #231

alexis-ld commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @carlito-scality

* @nicolas2bert

* @rachedbenmustapha

* @monpote

* @chengyanjin





The following reviewers are expecting changes from the author, or must review again:

* @chengyanjin







*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.0`




There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`




Please check the status of the associated issue None.

Goodbye alexis-ld.
